lib/iris/_data_manager.py

```diff
@@ -76,12 +79,69 @@ def __deepcopy__(self, memo):
         """
         return self._deepcopy(memo)
 
+    def __eq__(self, other):
+        """
+        Perform :class:`~iris._data_manager.DataManager` instance equality.
+        Note that, this is explicitly not a lazy operation and will load any
+        lazy payload to determine the equality result.
+
+        Comparison is strict with regards to lazy or real managed payload,
+        the realised_dtype, the dtype of the payload and the payload content.
+
+        Args:
+
+        * other:
+            The :class:`~iris._data_manager.DataManager` instance to
+            compare with.
+
+        Returns:
+            Boolean.
+
+        """
+        result = NotImplemented
+
+        if isinstance(other, type(self)):
+            result = False
+            is_lazy = self.has_lazy_data() == other.has_lazy_data()
```

Comment:
Trouble with this naming is that it suggests that, if `is_lazy` is True that both self and other have lazy data. But the following could be the case:
```
>>> print self.has_lazy_data()
False
>>> print other.has_lazy_data()
False
>>> is_lazy = self.has_lazy_data() == other.has_lazy_data()
>>> print is_lazy
True
```

What about `is_lazy_check` or `both_lazy`?